### PR TITLE
Gradle build optimization (harder, better, faster, stronger)

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

File: clustered/integration-test/src/test/java/org/ehcache/clustered/replication/BasicClusteredCacheOpsReplicationTest.java [hunk 1]
```diff
@@ -77,11 +82,16 @@
...
+  public static final ParallelTestCluster CLUSTER = new ParallelTestCluster(newCluster(2).in(new File("build/cluster")).withServiceFragment(RESOURCE_CONFIG).build());
+  @Rule
+  public final TestRule watchdog = CLUSTER.watchdog();
+
+  @Rule
+  public final TestName testName = new TestName();
 
   @Before
   public void startServers() throws Exception {
+    System.out.println("In @Before method");
```
Comment (mathieucarbou) [inline clustered/integration-test/src/test/java/org/ehcache/clustered/replication/BasicClusteredCacheOpsReplicationTest.java]: do we want to keep that ?

File: clustered/integration-test/src/test/java/org/ehcache/clustered/replication/BasicClusteredCacheOpsReplicationTest.java [hunk 2]
```diff
@@ -92,28 +102,28 @@ public void startServers() throws Exception {
...
+    cacheOne = cacheManager.createCache(testName.getMethodName() + "-1", config);
+    cacheTwo = cacheManager.createCache(testName.getMethodName() + "-2", config);
   }
 
   @After
-  public void tearDown() throws Exception {
-    CACHE_MANAGER.close();
-    CACHE_MANAGER.destroy();
+  public void tearDown() {
+    System.out.println("In @After method");
```
Comment (mathieucarbou) [inline clustered/integration-test/src/test/java/org/ehcache/clustered/replication/BasicClusteredCacheOpsReplicationTest.java]: sout

File: clustered/integration-test/src/test/java/org/ehcache/clustered/replication/BasicClusteredCacheOpsReplicationTest.java (hunk 1 above)
Comment (chrisdennis) [reply clustered/integration-test/src/test/java/org/ehcache/clustered/replication/BasicClusteredCacheOpsReplicationTest.java]: Nope... left over from my debugging. Will clean up!

File: clustered/integration-test/src/test/java/org/ehcache/clustered/sync/PassiveSyncTest.java [hunk 3]
```diff
@@ -89,10 +87,11 @@ public void testSync() throws Exception {
       CLUSTER.getClusterControl().terminateActive();
       CLUSTER.getClusterControl().waitForActive();
 
-      // Sometimes the new passive believes there is a second connection and we have to wait for the full reconnect window before getting a result
-      waitOrTimeout(() -> "value-5".equals(cache.get(-5L)), timeout(seconds(130)));
+      for (long end = System.nanoTime() + TimeUnit.SECONDS.toNanos(130); cache.get(0L) == null && System.nanoTime() < end; ) {
```
[the remark this answers is not in the document]
Comment (chrisdennis) [reply clustered/integration-test/src/test/java/org/ehcache/clustered/sync/PassiveSyncTest.java]: I'll take this over a test dependency for this one use.

File: clustered/integration-test/src/test/java/org/ehcache/clustered/util/ParallelTestCluster.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  public TestRule watchdog() {
+    return new TestWatcher() {
+
+      @Override
+      protected void starting(Description description) {
+        membership.register();
+        try {
+          Thread.sleep(100);
```
Comment (mathieucarbou) [inline clustered/integration-test/src/test/java/org/ehcache/clustered/util/ParallelTestCluster.java]: why we need a sleep ?

Comment (chrisdennis) [reply clustered/integration-test/src/test/java/org/ehcache/clustered/util/ParallelTestCluster.java]: We don't *need* a sleep... we sleep to give some time for more participants to show up for the test cycle.

File: clustered/integration-test/src/test/java/org/ehcache/clustered/sync/PassiveSyncTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lorban) [reply clustered/integration-test/src/test/java/org/ehcache/clustered/sync/PassiveSyncTest.java]: I wouldn't. But I'll leave that up to you.

**Pull request merged**